### PR TITLE
Abdm facility and version 3 changes

### PR DESCRIPTION
**Pull request by helenKaryamsetty**
## 📋 Description

JIRA ID: AMM-854, 1017,1016

Please provide a summary of the change and the motivation behind it. Include relevant context and details.

---

## ✅ Type of Change

- [ ] 🐞 **Bug fix** (non-breaking change which resolves an issue)
- [ ] ✨ **New feature** (non-breaking change which adds functionality)
- [ ] 🔥 **Breaking change** (fix or feature that would cause existing functionality to not work as expected)
- [ ] 🛠 **Refactor** (change that is neither a fix nor a new feature)
- [ ] ⚙️ **Config change** (configuration file or build script updates)
- [ ] 📚 **Documentation** (updates to docs or readme)
- [ ] 🧪 **Tests** (adding new or updating existing tests)
- [ ] 🎨 **UI/UX** (changes that affect the user interface)
- [ ] 🚀 **Performance** (improves performance)
- [ ] 🧹 **Chore** (miscellaneous changes that don't modify src or test files)

---

## ℹ️ Additional Information

Please describe how the changes were tested, and include any relevant screenshots, logs, or other information that provides additional context.

## Release Notes

- **New Features**
	- Enhanced functionality for managing health facilities linked to patient visits.
	- Improved user interface localization for ABHA-related processes in Assamese, English, and Hindi.

- **Bug Fixes**
	- Clarified button actions in the dialog component to enhance user experience.

- **Chores**
	- Updated environment configurations to reflect changes in ABHA extension settings.

### CODE REVIEW

File: src/app/app-modules/nurse-doctor/workarea/workarea.component.ts [hunk 1]
```diff
@@ -3237,6 +3240,61 @@ export class WorkareaComponent
...
+      const roles = jsonLoccationData?.previlegeObj[0]?.roles;
+      roles.find((item: any) => {
+        item.RoleName.toLowerCase() === 'doctor';
+        workLocationId = item.workingLocationID;
+      });
+    }
+    console.log('workLocationId', workLocationId);
+    this.registrarService.getMappedFacility(workLocationId).subscribe(
+      (res: any) => {
+        if (res.statusCode === 200 && res.data != null) {
```
Comment (devikasuresh20) [inline src/app/app-modules/nurse-doctor/workarea/workarea.component.ts]: Please use  "!==" instead of "!="

**Review by devikasuresh20 — CHANGES_REQUESTED** [review]

File: src/environments/environment.dev.ts [hunk 2]
```diff
@@ -66,7 +66,7 @@ export const environment = {
   RBSTest: `RBS Test`,
   visualAcuityTest: `Visual Acuity Test`,
   haemoglobinTest: `Haemoglobin Test`,
-  abhaExtension: `@abdm`,
+  abhaExtension: `@sbx`,
```
[the remark this answers is not in the document]
Comment (helenKaryamsetty) [reply src/environments/environment.dev.ts:69]: @coderabbitai i have added those values based on environments... in dev environment it is sbx and prod environment it is abdm

File: src/app/app-modules/nurse-doctor/workarea/workarea.component.ts [hunk 3]
```diff
@@ -3237,6 +3240,61 @@
...
+    const jsonLoccationData = JSON.parse(locationData);
+    let workLocationId: any;
+    if (jsonLoccationData?.previlegeObj[0]?.roles) {
+      const roles = jsonLoccationData?.previlegeObj[0]?.roles;
+      roles.find((item: any) => {
+        item.RoleName.toLowerCase() === 'doctor';
+        workLocationId = item.workingLocationID;
+      });
+    }
+    console.log('workLocationId', workLocationId);
```
[the remark this answers is not in the document]
Comment (helenKaryamsetty) [reply src/app/app-modules/nurse-doctor/workarea/workarea.component.ts:3255]: I dont need a booelan keyword but data is needed and i want to use find() method only for my code purpose

File: src/app/app-modules/nurse-doctor/workarea/workarea.component.ts [hunk 4]
```diff
@@ -178,6 +178,8 @@
   isDoctorUpdate = false;
   isDoctorSave = false;
   serviceType: any;
+  abdmFacilityId = null;
+  abdmFacilityName = null;
```
[the remark this answers is not in the document]
Comment (helenKaryamsetty) [reply src/app/app-modules/nurse-doctor/workarea/workarea.component.ts:182]: Its maintained like that all over the project. hence following the same

File: src/app/app-modules/nurse-doctor/workarea/workarea.component.ts [hunk 5]
```diff
@@ -3237,6 +3240,61 @@ export class WorkareaComponent
...
   }
+
+  getMappedAbdmFacility() {
+    const locationData: any = localStorage.getItem('loginDataResponse');
+    const jsonLoccationData = JSON.parse(locationData);
+    let workLocationId: any;
+    if (jsonLoccationData?.previlegeObj[0]?.roles) {
+      const roles = jsonLoccationData?.previlegeObj[0]?.roles;
+      roles.find((item: any) => {
+        item.RoleName.toLowerCase() === 'doctor';
```
Comment (drtechie) [inline src/app/app-modules/nurse-doctor/workarea/workarea.component.ts:3251]: comparison?

Comment (helenKaryamsetty) [reply src/app/app-modules/nurse-doctor/workarea/workarea.component.ts:3251]: @drtechie Yes, we are fetching the role mapped to userName and only giving carecontext facility details if the role is doctor.

**Review by devikasuresh20 — APPROVED** [review]

**Pull request merged**